packages/builder/src/pages/builder/app/[application]/data/datasource/[selectedDatasource]/CreateEditRelationship/CreateEditRelationship.svelte

```diff
@@ -0,0 +1,210 @@
+        ...relateFrom,
+        through: through._id,
+        fieldName: toTable.primary[0],
+      }
+      relateTo = {
+        ...relateTo,
+        through: through._id,
+        fieldName: fromTable.primary[0],
+      }
+    } else {
+      relateFrom = {
+        ...relateFrom,
+        foreignKey: relateFrom.fieldName,
+        fieldName: fromTable.primary[0],
+      }
+      relateTo = {
+        ...relateTo,
+        relationshipType: RelationshipTypes.ONE_TO_MANY,
+        foreignKey: relateFrom.fieldName,
+        fieldName: fromTable.primary[0],
+      }
+    }
+
+    fromRelationship = relateFrom
+    toRelationship = relateTo
+  }
+
+  // save the relationship on to the datasource
+  async function saveRelationship() {
+    buildRelationships()
+    // source of relationship
+    datasource.entities[fromTable.name].schema[fromRelationship.name] =
+      fromRelationship
+    // save other side of relationship in the other schema
+    datasource.entities[toTable.name].schema[toRelationship.name] =
+      toRelationship
+
+    // If relationship has been renamed
+    if (originalFromName !== fromRelationship.name) {
+      delete datasource.entities[fromTable.name].schema[originalFromName]
+    }
+    if (originalToName !== toRelationship.name) {
+      delete datasource.entities[toTable.name].schema[originalToName]
+    }
+
+    await save()
+    await tables.fetch()
+  }
+
+  async function deleteRelationship() {
+    delete datasource.entities[fromTable.name].schema[fromRelationship.name]
+    delete datasource.entities[toTable.name].schema[toRelationship.name]
+    await save()
+    await tables.fetch()
+    close()
+  }
+</script>
+
+<ModalContent
+  size="XL"
+  title="Create Relationship"
+  confirmText="Save"
+  onConfirm={saveRelationship}
+  disabled={!valid}
+>
+  <div class="relationship-names">
+    <div class="left-name">
+      <Input label="From name" bind:value={fromRelationship.name} />
+    </div>
+    <div class="right-name">
+      <Input label="To name" bind:value={toRelationship.name} />
+    </div>
+  </div>
+
+  <div class="table-selector">
+    <Select
+      label="Relationship"
+      options={relationshipTypes}
+      bind:value={fromRelationship.relationshipType}
+    />
+
+    <Select
+      label="From"
+      options={tableOptions}
+      bind:value={toRelationship.tableId}
+    />
+
+    <Select
+      label={"Has many"}
+      options={tableOptions}
+      bind:value={fromRelationship.tableId}
+    />
+
+    {#if fromRelationship?.relationshipType === RelationshipTypes.MANY_TO_MANY}
+      <Select
+        label={"Through"}
+        options={tableOptions}
+        bind:value={fromRelationship.through}
+      />
+    {:else if toTable}
+      <Select
+        label={`Foreign Key (${toTable?.name})`}
+        options={Object.keys(toTable?.schema)}
+        bind:value={fromRelationship.fieldName}
+      />
+    {/if}
+  </div>
+
+  <div slot="footer">
+    {#if originalFromName !== null}
```

Comment:
Just need to swap this to `!=` to catch `undefined` - I noticed the delete button was still visible when creating a relationship and the extra equals here is the culprit.